gm4_block_compressors/data/gm4_block_compressors/functions/item.mcfunction

```diff
@@ -5,9 +5,30 @@
+execute store success score @s gm4_bc_data if data storage gm4_block_compressors:temp/compressor {Items:[{Count:1b}]} if data storage gm4_block_compressors:temp/item_stack Item{Count:1b,tag:{gm4_compressed:{}}} run function gm4_block_compressors:item/decompress
+execute if data storage gm4_block_compressors:temp/item_stack Item{Count:1b} run scoreboard players set @s gm4_bc_data 2
+
+
+execute unless score @s gm4_bc_data matches 2 if data storage gm4_block_compressors:temp/compressor Items[0] store success score @s gm4_bc_data run data modify storage gm4_block_compressors:temp/compressor Items[0].Count set from storage gm4_block_compressors:temp/item_stack Item.Count
+execute if score @s gm4_bc_data matches 1 if data storage gm4_block_compressors:temp/compressor Items[1] store success score @s gm4_bc_data run data modify storage gm4_block_compressors:temp/compressor Items[1].Count set from storage gm4_block_compressors:temp/item_stack Item.Count
+execute if score @s gm4_bc_data matches 1 if data storage gm4_block_compressors:temp/compressor Items[2] store success score @s gm4_bc_data run data modify storage gm4_block_compressors:temp/compressor Items[2].Count set from storage gm4_block_compressors:temp/item_stack Item.Count
+execute if score @s gm4_bc_data matches 1 if data storage gm4_block_compressors:temp/compressor Items[3] store success score @s gm4_bc_data run data modify storage gm4_block_compressors:temp/compressor Items[3].Count set from storage gm4_block_compressors:temp/item_stack Item.Count
+execute if score @s gm4_bc_data matches 1 if data storage gm4_block_compressors:temp/compressor Items[4] store success score @s gm4_bc_data run data modify storage gm4_block_compressors:temp/compressor Items[4].Count set from storage gm4_block_compressors:temp/item_stack Item.Count
+execute if score @s gm4_bc_data matches 1 if data storage gm4_block_compressors:temp/compressor Items[5] store success score @s gm4_bc_data run data modify storage gm4_block_compressors:temp/compressor Items[5].Count set from storage gm4_block_compressors:temp/item_stack Item.Count
+execute if score @s gm4_bc_data matches 1 if data storage gm4_block_compressors:temp/compressor Items[6] store success score @s gm4_bc_data run data modify storage gm4_block_compressors:temp/compressor Items[6].Count set from storage gm4_block_compressors:temp/item_stack Item.Count
+execute if score @s gm4_bc_data matches 1 if data storage gm4_block_compressors:temp/compressor Items[7] store success score @s gm4_bc_data run data modify storage gm4_block_compressors:temp/compressor Items[7].Count set from storage gm4_block_compressors:temp/item_stack Item.Count
+execute if score @s gm4_bc_data matches 1 if data storage gm4_block_compressors:temp/compressor Items[8] store success score @s gm4_bc_data run data modify storage gm4_block_compressors:temp/compressor Items[8].Count set from storage gm4_block_compressors:temp/item_stack Item.Count
```

Comment:
What is this for? The NBT `storage gm4_block_compressors:temp/compressor Items[8].Count` gets set here but the module doesn't seem to read this NBT anywhere, so why set it in the first place?

Reply:
This is how I do the comparisons.
The `if score` makes sure it only runs until it finds a match, and doesn't overwrite the match if later ones do not match.
The `if data` checks if the that slot exists, as otherwise it'll fail regardless.
The actual comparison is done by the `store success` and the `data modify`. If you try to copy a nbt value, but the original value is the same as the new value, you it will fail, and you'll get a success of 0.  If it ever gets set to zero, the following steps get skipped, and it'll run `compress.mcfunction`

An alternative way of doing this would be to copy the item's `Count` to a score, copy each slot's `Count` to a score, then to compare each of those scores with the items score. I'm not sure which would be better.

Breakdown of the number of commands used for each method in the worst case (no matches)

Example comparison for both
NBT:
```
execute if score action gm4_bc_data matches 1 if data storage gm4_block_compressors:temp/compressor Items[1] store success score action gm4_bc_data run data modify storage gm4_block_compressors:temp/compressor Items[1].Count set from storage gm4_block_compressors:temp/item_stack Item.Count
```
1 `if score ... matches`, 1 `if data`, 1 `score success score`, 1 `data merge ... from ...`

Scores
```
execute unless score action gm4_bc_data matches 1 store result score slot_1 gm4_bc_data run data get storage gm4_block_compressors:temp/compressor Items[0].Count
execute unless score action gm4_bc_data matches 1 if score slot_1 gm4_bc_data = item_count gm4_bc_data run function gm4_block_compressors:item/compress
```
2 `if score ... matches`, 1 `if score ... = ...`, 1 `score result score`, 1 `data get`

Reply:
So this is to ensure all stacks inside the compressors are the same size?
I think previously compressors just looked at the last item, which is very useful.

Reply:
This will check if any item in the compressor matches the stack size of the item. It checks if it can copy the stacksize from the item to slot n. If that copy fails, it knows that both the item and slot n have the same count, and stops further checks.


The current version does the same thing. The current version allows you to put 64, 16, and 1 item into the same compressor, and it'll be able to do all 3 actions, no matter where those items are in the compressor. It can do this using `Items[{Count:64b}]`, which checks if any slot has 64 items. I can't do that because the 64 has to be dynamic, so direct comparison of each one slot is the best that can be done as far as I know.